crates/dojo/core/src/world/world_contract.cairo

```diff
@@ -953,6 +950,45 @@ pub mod world {
             }
         }
 
+        /// Panics if a resource is not upgradable.
+        ///
+        /// Upgradable means:
+        /// - the version of the resource must be incremented by 1,
+        /// - the layout type must remain the same (Struct or Fixed),
+        /// - existing fields cannot be changed or moved inside the resource,
+        /// - new fields can only be appended at the end of the resource.
+        ///
+        /// # Arguments
+        ///   * `selector` - the selector of the resource.
+        ///   * `prev_address` - the address of the current resource.
+        ///   * `new_address` - the address of the newly deployed resource.
+        ///
+        fn assert_resource_upgradability(selector: felt252, prev_address: address, new_address: address) {
```

Comment:
We should have the namespace as input + we have access to the `name` using the `IDeployedResource`. Like this, the error message will be clean with name and namespace in clear.